test/support/repo.ex

```diff
@@ -1,3 +1,4 @@
 defmodule EctoJob.Test.Repo do
-  use Ecto.Repo, otp_app: :ecto_job, adapter: Ecto.Adapters.Postgres
+  adapter = Application.compile_env!(:ecto_job, __MODULE__)[:adapter]
```

Comment:
`Application.compile_env!` isn't available in elixir 1.9, causing a failure in Travis CI.

Would you mind updating the `.travis.yml` file to only test on elixir 1.10.4 with OTP versions 21 and 23?
We can add elixir after it is released next month.

We should probably exercise the MySQL support in CI too - can be a separate PR if you prefer.